crates/red_knot_python_semantic/src/semantic_index/builder.rs

```diff
@@ -416,6 +424,22 @@ where
                     self.flow_merge(&pre_if);
                 }
             }
+            ast::Stmt::While(node) => {
+                self.visit_expr(&node.test);
+                let pre_loop = self.flow_snapshot();
+                let mut break_states: Vec<FlowSnapshot> = vec![];
+                std::mem::swap(&mut self.loop_break_states, &mut break_states);
+                self.visit_body(&node.body);
+                std::mem::swap(&mut self.loop_break_states, &mut break_states);
+                self.flow_merge(&pre_loop);
+                self.visit_body(&node.orelse);
+                for break_state in break_states {
+                    self.flow_merge(&break_state);
+                }
```

Comment:
Nit: Some grouping of the statements could help readability
```suggestion
                self.visit_expr(&node.test);
                
                let pre_loop = self.flow_snapshot();
                let mut break_states: Vec<FlowSnapshot> = vec![];
                std::mem::swap(&mut self.loop_break_states, &mut break_states);
                self.visit_body(&node.body);
                std::mem::swap(&mut self.loop_break_states, &mut break_states);
                self.flow_merge(&pre_loop);
                
                self.visit_body(&node.orelse);
                for break_state in break_states {
                    self.flow_merge(&break_state);
                }
```